search/median_search2.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file
+ *
+ * @details
```

Comment:
Could you add a brief description of the algorithm, as well as a Wikipedia link to the algorithm?


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not maintainable

Reply:
I have made the requested changes, please review again.